lib/plugins/sfDoctrinePlugin/lib/generator/sfDoctrineFormGenerator.class.php

```diff
@@ -554,44 +553,41 @@ public function underscore($name)
     return strtolower(preg_replace(array('/([A-Z]+)([A-Z][a-z])/', '/([a-z\d])([A-Z])/'), '\\1_\\2', $name));
   }
 
-  /**
-   * Get array of sfDoctrineColumn objects that exist on the current model but not its parent.
-   *
-   * @return array $columns
-   */
-  public function getColumns()
-  {
-    $parentModel = $this->getParentModel();
-    $parentColumns = $parentModel ? array_keys(Doctrine_Core::getTable($parentModel)->getColumns()) : array();
-
-    $columns = array();
-    foreach (array_diff(array_keys($this->table->getColumns()), $parentColumns) as $name)
+    /**
+     * Get array of sfDoctrineColumn objects that exist on the current model but not its parent.
+     *
+     * @return array $columns
+     */
+    public function getColumns()
     {
-      $columns[] = new sfDoctrineColumn($name, $this->table);
-    }
+        $parentModel = $this->getParentModel();
+        $parentColumns = $parentModel ? array_keys(Doctrine_Core::getTable($parentModel)->getColumns()) : [];
 
-    // add relations to columns for inherited classes
-    if ($parentModel)
-    {
-      $parentRelationNames = array_keys(Doctrine_Core::getTable($parentModel)->getRelations());
-      $relations = $this->table->getRelations();
-      $relationColumns = array();
-      foreach (array_diff(array_keys($relations), $parentRelationNames) as $relationName)
-      {
-        if (Doctrine_Relation::ONE == $relations[$relationName]->getType())
-        {
-          $columnName = $relations[$relationName]->getLocal();
-          if (!in_array($columnName, $relationColumns))
-          {
-            $relationColumns[] = $columnName;
-            $columns[] = new sfDoctrineColumn($columnName, $this->table);
-          }
```

Comment:
At this point in the method, the `$columns` variable already contains each of the first-party properties of the model, but now we're looping through the _relationships_ to see if any other properties need to be added.

The test being done only looks to make sure that a _relationship_ for the current column hasn't already been added to `$columns`. It **doesn't** look to see if that column already exists in `$columns` as a first-party property. This means that **any model** which contained the `one` side of a `one-to-many` relationship would have duplication of that property in its base classes.